hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordScanner.java

```diff
@@ -202,23 +202,21 @@ public void scan() {
                     LOG.info("Rolling back the last corrupted log block read in " + logFile.getPath());
                     currentInstantLogBlocks.pop();
                     numBlocksRolledBack++;
-                  } else if (lastBlock.getBlockType() != CORRUPT_BLOCK
```

Comment:
Can you explain what’s changing here in terms of corrupt block handling

Reply:
I have simplified the if-else conditions. 

if checks for (lastBlock.getBlockType() == CORRUPT_BLOCK) so the else does not need to check for the != part (which should be true if we reach the else part.